packages/core-graph/src/private/schemas/colorNode.ts

```diff
@@ -0,0 +1,161 @@
+import type { FlatNodeObject } from './FlatNodeObject';
+import { IdentifierSchema } from './Identifier';
+import { JSONLiteralSchema, type JSONLiteral } from './JSONLiteral';
+import { LiteralSchema, type Literal } from './Literal';
+import { NodeReferenceSchema, type NodeReference } from './NodeReference';
+
+// Our opinions.
+const SlantNodeSchema = pipe(
+  objectWithRest(
+    {
+      // We treat @context as opaque string than a schema.
+      '@context': optional(string('@context must be an IRI')),
+      '@id': IdentifierSchema,
+      // Multi-membership is enabled by default.
+      '@type': pipe(
+        array(string('element in @type must be a string'), '@type must be array of string'),
+        freeze(),
+        minLength(1, '@type must have at least one element')
+      ),
+      // We follow Schema.org that "hasPart" denotes children.
+      // This relationship is "membership" than "hierarchy".
+      hasPart: optional(
+        pipe(
+          array(NodeReferenceSchema, 'hasPart must be array of NodeReference'),
+          freeze(),
+          minLength(1, 'hasPart, if present, must have at least one element')
+        )
+      ),
+      // We follow Schema.org that "isPartOf" denotes parent, and multiple parent is possible.
+      // This relationship is "membership" than "hierarchy".
+      isPartOf: optional(
+        pipe(
+          array(NodeReferenceSchema, 'isPartOf must be array of NodeReference'),
+          freeze(),
+          minLength(1, 'isPartOf, if present, must have at least one element')
+        )
+      )
+    },
+    // The rest property values must be encapsulated in array.
+    // Array of boolean, number, string, JSON literal, and node reference are accepted.
+    pipe(
+      array(
+        union(
+          [JSONLiteralSchema, LiteralSchema, NodeReferenceSchema],
+          'Properties of slant node must be array of JSON literal, literal or node reference'
+        )
+      ),
+      minLength(1, 'Properties of slant node must be an array with at least 1 element')
+    )
+  ),
+  freeze()
+);
+
+// Due to limitation on TypeScript, we cannot truthfully represent the typing.
+type SlantNode = InferOutput<ObjectSchema<(typeof SlantNodeSchema)['entries'], undefined>> & {
+  [key: string]: unknown;
+};
+
+const SlantNodeWithFixSchema = pipe(
+  looseObject({}),
+  transform(node => {
+    const propertyMap = new Map<string, readonly (JSONLiteral | Literal | NodeReference)[]>();
+    let context: string | undefined;
+    let id: string | undefined;
+
+    for (const [key, value] of Object.entries(node)) {
+      const parsedValue = parse(
+        union(
+          [
+            array(union([JSONLiteralSchema, LiteralSchema, NodeReferenceSchema])),
+            JSONLiteralSchema,
+            LiteralSchema,
+            NodeReferenceSchema,
+            null_()
+          ],
+          'Only JSON literal, literal, node reference or null can be parsed into slant node'
+        ),
+        value
+      );
```

Comment:
Let's extract this to the module scope, so we don't recreate the schema for each entry.